lib/hawkular/inventory/entities.rb

```diff
@@ -1,308 +1,121 @@
+    attr_reader :feed
+    # @return [ResourceType] Type of this resource
+    attr_reader :type
+    # @return [String] Parent ID of this entity (nil if it's a root resource)
+    attr_reader :parent_id
+    # @return [Hash<String,String>] Properties of this resource
+    attr_reader :properties
+    # @return [Hash<String,String>] Config map of this resource
+    attr_reader :config
+    # @return [List<Metric>] Metrics associated to this resource
+    attr_reader :metrics
+    # @return [List<Resource>] List of children (present when the whole tree is loaded, else nil)
+    attr_reader :children
 
     def initialize(hash)
-      @tenant_id = hash[:tenant_id]
-      @feed_id = hash[:feed_id]
-      @environment_id = hash[:environment_id]
-      @resource_type_id = hash[:resource_type_id]
-      @metric_type_id = hash[:metric_type_id]
-      @resource_ids = hash[:resource_ids]
-      @metric_id = hash[:metric_id]
-    end
-
-    def self.parse(path)
-      fail Hawkular::ArgumentError, 'CanonicalPath must not be nil or empty' if path.to_s.strip.length == 0
-
-      CanonicalPath.new(path_to_h path)
-    end
-
-    def self.parse_if_string(path)
-      path.is_a?(CanonicalPath) ? path : CanonicalPath.parse(path)
-    end
-
-    def copy_hash
-      hash = to_h
-      hash[:resource_ids] = hash[:resource_ids].clone unless hash[:resource_ids].nil?
-      hash
-    end
-
-    # Move up to the parent path of the resource. resource_ids set to empty array when there is no parent.
-    # @return CanonicalPath corresponding to the direct ancestor of the resource represented by this path object.
-    def up
-      hash = copy_hash
-      res = hash[:resource_ids] || []
-      res = res.take(res.length - 1) unless res.empty?
-      hash[:resource_ids] = res
-      CanonicalPath.new(hash)
-    end
-
-    # Add resource down to the current path
-    # @return a new CanonicalPath based on the current one
-    def down(resource)
-      hash = copy_hash
-      hash[:resource_ids] = (hash[:resource_ids] || []) << hawk_escape_id(resource)
-      CanonicalPath.new(hash)
+      @id = hash['id']
+      @name = hash['name']
+      @feed = hash['feedId']
+      @type = ResourceType.new(hash['type'])
+      @parent_id = hash['parentId']
+      @properties = hash['properties'] || {}
+      @config = hash['config'] || {}
+      @metrics = (hash['metrics'] || []).map { |m| Metric.new(m) }
+      @children = hash['children'].map { |r| Resource.new(r) } if hash.key? 'children'
+      @_hash = hash.dup
     end
 
-    # Set resource type to the current path
-    # @return a new CanonicalPath based on the current one
-    def resource_type(resource_type)
-      hash = copy_hash
-      hash[:resource_type_id] = hawk_escape_id(resource_type)
-      CanonicalPath.new(hash)
+    def children(recursive = false)
+      return @children unless recursive == true
+      fail Hawkular::ArgumentError 'Resource tree not loaded, load it by calling resource_tree' if @children.nil?
```

Comment:
I don't see this as an argument error.

Reply:
I'm open to suggestion/ideas :)

Is something like an `Invalid Call` `Call Error`, `ResourceTreeNotLoadedException` ?

Reply:
For information, this is not something that was added in this PR. This PR focuses on removing old inventory, renaming new inv from "v4" to just "inventory" and starting fixing tests, but github diff shows differences between old and renamed files. I think it causes a lot of misunderstanding about this PR. I would have liked it was merged very fast so that we can go on from this basis.